pandas/_libs/tslibs/period.pyx

```diff
@@ -1685,7 +1685,7 @@ cdef class _Period(object):
                 if other.freq != self.freq:
                     msg = _DIFFERENT_FREQ.format(self.freqstr, other.freqstr)
                     raise IncompatibleFrequency(msg)
-                return (self.ordinal - other.ordinal) * self.freq
+                return (self.ordinal - other.ordinal) * type(self.freq)()
```

Comment:
Won't this be wrong for offsets that have relevant keywords?

Reply:
You’re right. So one option would be to do `... * self.freq * (1. / self.freq.n)`. (Since DateOffset doesn’t support div) This way no other classes need to be modified. However, this could probably lead to unforeseen floating point issues, so I think it might be worth adding a property to the DateOffset objet to provide the object with n=1. Something like `DateOffset.base_freq`? Or __div__ that only allows factors of n.

Reply:
done